appcues/src/test/java/com/appcues/data/AppcuesRepositoryTest.kt

```diff
@@ -140,6 +140,10 @@ internal class AppcuesRepositoryTest {
         val activityStorage = ActivityStorage(UUID.randomUUID(), "123", "userId", "data", null)
         coEvery { appcuesLocalSource.getAllActivity() } returns listOf(activityStorage)
 
+        // two lines below added related to https://github.com/mockk/mockk/issues/321
+        coEvery { appcuesRemoteSource.postActivity(any(), any(), any()) } returns Success(mockk(relaxed = true))
+        coEvery { appcuesRemoteSource.qualify(any(), any(), any(), any()) } returns Success(mockk(relaxed = true))
```

Comment:
I had to update mockk, I assume for the Kotlin update, and without that many things failed.

However, it did introduce this weird issue in this class only where mocked return types with Generics have some unexpected behavior. In our case, its the ResultOf (Success/Failure) types used in the lower level network layer as the return value of `postActivity` and `qualify` here. It is basically exactly as noted in this issue https://github.com/mockk/mockk/issues/321 but this workaround to more explicitly set the mock response rather than using a relaxed mock works ok for now.